kedro/extras/datasets/pandas/sql_dataset.py

```diff
@@ -346,6 +350,10 @@ def __init__(  # pylint: disable=too-many-arguments
                 https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
                 All defaults are preserved, except `mode`, which is set to `r` when loading.
             filepath: A path to a file with a sql query statement.
+            execution_options: A dictionary with non-SQL options for the connection to
+                be applied to the underlying engine. To find all supported execution
+                options, see here:
+                https://docs.sqlalchemy.org/en/12/core/connections.html#sqlalchemy.engine.Connection.execution_options
```

Comment:
I think this deserves more documentation as this is an advance feature and not documented in the pandas API. 

i.e. What options are common and when do we use it?

The article discusses client-side & server-side computation, are there any tradeoffs here, or server-side is always preferred? I think we should at least mention that if users are running into memory issues, it is worths trying this option.
https://pythonspeed.com/articles/pandas-sql-chunking/